zoo_frontend/src/pages/kitchen/prep/kitchen.jsx

```diff
@@ -9,172 +11,151 @@ import NativeSelect from '@material-ui/core/NativeSelect';
-  async getKitchenData(speciesID, dietID, dcID) {
-    const [species, prepNotes, dc, dietChanges] = await Promise.all([
-      this.serverSpeciesAPI.getSpecies({
-        where: {
-          speciesId: speciesID,
-        },
-      }),
-      this.serverPrepNotesAPI.getPrepNotes({
-        where: {
-          dietId: dietID,
-        },
-      }),
-      this.serverDeliverContainersAPI.getDeliveryContainers({
-        where: {
-          dcId: dcID,
-        },
-      }),
-      this.serverDietChangesAPI.getDietChanges({
-        where: {
-          dietId: dietID,
-        },
-      }),
-    ]);
-
-    this.setState({
-      species: species.data[0].species,
-      prepNotes: prepNotes.data,
-      dc: dc.data[0].dc,
-      dietChanges: dietChanges.data.slice(0, 3),
-      foodPrep: this.getPrepFood(dietID),
-    });
-  }
+    let tableIndex = tables.findIndex((table) => Number(this.props.table || -1) === table.tableId);
+
+    if(tableIndex === -1) {
+      tableIndex = 0;
+    }
+
+    let currentIndex = tables[tableIndex].diets.findIndex((diet) => Number(this.props.diet || 0) === diet.diet_id);
 
-  /* Returns the food prep for the diet/animal */
-  // For entry in dietsSub where diet_id = dietID
-  // food, group_amount
-  getPrepFood(dietID) {
-    const items = this.props.PrepDietsSub.filter((item) => item.diet_id === dietID);
-    return items;
+    if(currentIndex === -1) {
+      currentIndex = 0;
+    }
+
+    this.state = {
+      tableIndex,
+      currentIndex,
+      tables,
+    };
   }
 
   /* Handle table change !! */
-  handleChange = (name) => (event) => {
-    const e = event.target.value;
-    this.setState({ [name]: e, currentIndex: 0 });
-    this.getDietsData(e).then(() => {
-      this.updateState();
-    });
+  handleChange = (event) => {
+    const value = Number(event.target.value);
+    this.setState(({ tables }) => ({
+      tableIndex: tables.findIndex((table) => table.tableId === value),
+      currentIndex: 0,
+    }), this.updateUrl);
   };
 
   handleNext = () => {
     this.setState((prevState) => ({
       currentIndex: prevState.currentIndex + 1,
-    }));
-    this.updateState();
+    }), this.updateUrl);
   };
 
   handlePrev = () => {
     this.setState((prevState) => ({
       currentIndex: prevState.currentIndex - 1,
-    }));
-    this.updateState();
+    }), this.updateUrl);
   };
 
-  updateState = () =>
-    this.getKitchenData(
-      this.state.diets[this.state.currentIndex].speciesId, // speciesID
-      this.state.diets[this.state.currentIndex].dietId, // dietID
-      this.state.diets[this.state.currentIndex].dcId, // dcID
-      this.props.date,
-    );
+  updateUrl = () => {
+    const table = this.state.tables[this.state.tableIndex];
+    
+    let route = `${Kitchen.prep.link}?date=${this.props.date}&table=${table.tableId}`
+
+    if(table.diets.length > this.state.currentIndex) {
+      route += `&diet=${table.diets[this.state.currentIndex].diet_id}`
+    }
+
+    Router.push(route);
```

Comment:
ah here's your problem, you can .push without redirecting and thus calling `getInitialProps` from the server. 
if you do: `Router.push(href, href, { shallow: true });` the shallow option is what you need here.

Reply:
I think I had that then changed it good catch